### PR TITLE
Use ccache on Travis to speedup installation

### PR DESCRIPTION
**Pull request by saimn**
This should gain a few minutes of compilation time. It is handled directly by Travis (https://docs.travis-ci.com/user/caching/#ccache-cache). The cache must first be filled, then we can run Travis again to see the gain.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,9 @@ language: c
 
 compiler: gcc
 
+cache:
```
Comment (astrofrog) [inline .travis.yml]: Can you add a comment with a mention of how to reset the cache if needed?

**Comment by astrofrog** [comment]
I've restarted Travis. Initial timings:

screen shot 2017-11-29 at 18 27 19

**Comment by astrofrog** [comment]
Hmm although there is a lot of variation between builds:

screen shot 2017-11-29 at 18 28 58

**Comment by bsipocz** [comment]
Well, it's almost 20 minutes, we haven't gone under an hour total for a long long time. 🎉

**Comment by bsipocz** [comment]
Thanks @saimn, this is a very significant improvement!

**Comment by saimn** [comment]
Yes, sometimes travis builds are significantly slower (and this is why doing the benchmarks on Travis would be difficult), but overall the gain seems stable and consistent with my tests, around 2min30s per build.

For possible future improvements, there is also the possibility to cache the pip directory (.tar.gz and wheels), but I'm not sure the gain would be significant: Travis also download the cache from another host, so compared to getting wheels from pypi it's not obvious; and anyway most packages come from conda. Maybe something to try to be sure.

**Pull request merged**